CHANGELOG.md

```diff
@@ -0,0 +1,36 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
+
+## [Unreleased]
+
+### Added
+
+- Added `n_points` parameter to all functions in `shap.datasets` (#39).
+- Added the ruff linter (#25, #26, #27).
+
+### Fixed
+
+- Fixed installation of package via setuptools (#51).
+- Fixed failing unit tests (#29).
+- Fixed deprecation warnings from `numpy` types (#7).
+- Fixed deprecation warnings from `Ipython.core.display` (#13).
+- Fixed deprecation warnings from `tensorflow` optimisers (#16).
+- Fixed deprecation warnings from `sklearn.linear_model` (#22).
```

Comment:
Thanks for the review comments, I'll push an update when I get some time.

> Thanks for the initiative. Btw, let's agree that we will only update this file before release, by one of the maintainers. Rather than enforce it in every PR yea? It would become too messy otherwise

I generally prefer advocate for adding in items to an "Unreleased" section on an ongoing basis, as recommended in [How can I reduce the effort required to maintain a changelog](https://keepachangelog.com/en/1.0.0/#effort). Benefits of an incremental approach are:
1. We don't accrue too much tech debt having to review diffs on each release, and
2. The changelog remains reasonably up-to-date at all times, without waiting for a release.

I might be missing something, what is the messiness that you allude to with this approach? The risk of annoying merge conflicts can be mitigated quite easily: we can add `CHANGELOG merge=union` to a `.gitattributes`, so that multiple additions are automatically concatenated together without raising a conflict.

Certainly, some PRs from new contributors might omit an entry to the changelog, so we will probably always need to do a bit of cleanup before making a release. I'd rather not have to gather it all from scratch each time though.

Reply:
> we can add CHANGELOG merge=union to a .gitattributes, so that multiple additions are automatically concatenated together without raising a conflict.

Cool, I didn't know about this. The problem I had with my previous OSS project was that we would have a changelog.md, that in 1 PR looks like

```
- introduced xxx in #101
- introduced yyy in #102
- introduced zzz in #104
```

and in another that looks like 
```
- introduced xxx in #101
- introduced www in #103
- introduced zzz in #104
```
And merging PR 102 would then invalidate 103 because of a merge conflict.
Does setting the gitattributes resolve such problems?

If it does, then let's also add this change to this PR?

---

Agree with your pros. The above is really my only main concern.

I'll also add one last comment:

Based on what I've seen: probably most contributors won't add an entry to the CHANGELOG, with or without a contributing guideline / a reminder in the issue template. And I don't want to have to force this upon the contributors, 1) it adds one additional hurdle for them; and 2) IMO project admin should be handled by the maintainers, not the contributors. I'll rather they spend the time on writing good tests and making sure they pass. 3) There's also no guarantee that their changelog entry meets our changelog "style", e.g. grammatically correct, including the PR number.. so I'll much rather we control the changelog ourselves.

Reply:
Sounds good! Let's put the onus on maintainers, rather than on each PR.

This topic is a bit prone to [bikeshedding](https://en.wiktionary.org/wiki/bikeshedding), which I'll try to avoid. The main thing is that we get it done :)